app.js

```diff
@@ -22,14 +23,29 @@ app.post('/delta', bodyParser.json(), async (req, res) => {
       updateablePredicates.push(SUBCASE_ACTIVITY_PREDICATES[predicate]);
     }
   }
+  for (const predicate in ACTIVITY_PREDICATES) {
+    if (ACTIVITY_PREDICATES.hasOwnProperty(predicate)) {
+      updateablePredicates.push(ACTIVITY_PREDICATES[predicate]);
+    }
+  }
   const pathUpdates = deltaUtil.filterByPredicate(insertionDeltas, updateablePredicates);
   if (pathUpdates.length) {
     console.log(`Received deltas for ${pathUpdates.length} activities that might imply a change in status for their signflow.`);
   }
   for (const d of pathUpdates) {
-    const signSubcaseUri = d.object.value;
-    await syncStatusForSignSubcase(signSubcaseUri);
+    let signSubcaseUri;
+
+    const predicate = d.predicate.value;
+    if (Object.values(SUBCASE_ACTIVITY_PREDICATES).includes(predicate)) {
+      signSubcaseUri = d.object.value;
+    } else {
+      signSubcaseUri = await fetchSignSubcaseUri(d.subject.value);
+    }
```

Comment:
In this case, the predicate "dossier startdatum" could be 1 of many activities.
To solve this you verify the subject with a query.

Would it make sense to have a config with all possible sign flow activities prefixes and loop them to match the `d.subject.value` to known activities and proceed if found? the value should be the prefix + the id (I think)
`http://themis.vlaanderen.be/id/handteken-handtekenactiviteit/`.

Reply:
scratch that, You don't have the signSubcase yet...

Reply:
Had to look twice at that one